crates/core/src/client/mod.rs

```diff
@@ -631,6 +631,23 @@ impl KakarotClient for KakarotClientImpl<JsonRpcClient<HttpTransport>> {
         Ok(Address::from(evm_address_sliced))
     }
 
+    /// Get the nonce for a given ethereum address
+    /// Convert Ethereum address to Starknet address
+    /// Return the nonce, by calling `get_nonce` for the addresss via starknet rpc
+    /// * `ethereum_address` - The EVM address to get the nonce of
+    /// * `block_id` - The block to get the nonce at
+    ///
+    /// ### Returns
+    /// * `Result<U256, KakarotClientError>` - The nonce of the EVM address
+    async fn nonce(&self, ethereum_address: Address, block_id: StarknetBlockId) -> Result<U256, KakarotClientError> {
+        let starknet_address = self.compute_starknet_address(ethereum_address, &block_id).await?;
+
+        let nonce_felt = self.inner.get_nonce(block_id, starknet_address).await?;
+        let nonce = U256::from_be_bytes(nonce_felt.to_bytes_be());
```

Comment:
can't we make `nonce.into()` ?

Reply:
@greged93 has a [PR](https://github.com/kkrt-labs/kakarot-rpc/pull/180) for this.

So, we have decided, that we can merge this, then he can rebase and implement this for this conversion as well.